android/src/test/java/com/revenuecat/purchases/common/CommonKtTests.kt

```diff
@@ -2,16 +2,19 @@ package com.revenuecat.purchases.common
 
 
 import com.revenuecat.purchases.Purchases
-import org.spekframework.spek2.Spek
-import org.spekframework.spek2.style.specification.describe
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
 import java.net.URL
 import kotlin.test.assertEquals
 import kotlin.test.assertFailsWith
-import kotlin.test.expect
 
-object CommonKtTests : Spek({
-    describe("setProxyURLString") {
-        it("sets the proxyURL correctly from a valid URL") {
+internal class CommonKtTests {
+    @Nested
+    @DisplayName("Calling setProxyURLString")
+    inner class SetProxyURLString {
+        @Test
```

Comment:
this is interesting, we haven't been splitting up test cases into groups like this, right? I like it. 
a slight advantage that Spek had was that you'd be able to very quickly create setup and tear down code that was specific to each group, does this structure still allow for that? 

I'm asking out of sheer curiosity because we weren't really taking advantage of the grouped setup and tear down anyway

Reply:
So we are using JUnit 5 for these tests, in `purchases-android` we still use JUnit 4, which doesn't let us do that. In this particular test suite doesn't really make sense to use this inner class, but in `SkuDetailsMapperTests` for example, it makes sense also for organization of the results. Android Studio will show the divided by groups which is nice. We can also do `@BeforeEach` and `@AfterEach` in each of these inner classes. I set it up following this article https://medium.com/@boonkeat/android-unit-testing-with-junit5-d1b8f9c620b6, in case you want to check it out.

Reply:
does the @nested with @displayname gets us any added functionality here (given that there's only one "nested" section in the file?)?  i see it's advantage for readability/future refactoring, but curious if there's any functionality or output associated with it..

Reply:
@vegaro posted that comment before seeing your response, feel free to ignore

Reply:
it shows the "nested" as a group, but being the single group, I don't see any benefit, so I am going to remove it. It would show similar to this:


![](https://miro.medium.com/max/700/1*mG9KiB3UfDm-bMUOV2Vr0w.png)